onnx/test/cpp/data_propagation_test.cc

```diff
@@ -18,37 +18,29 @@ namespace ONNX_NAMESPACE {
 
 namespace Test {
 
-inline bool CompareShape(const TensorShapeProto& A, const TensorShapeProto& B,
-  bool checkSameParam = false) {
-    if (A.dim_size() != B.dim_size()) {
-        fail_check("The compared sizes of dim are different.");
-        return false;
-    }
-    for (int i = 0; i < A.dim_size() ; ++i) {
-      if (A.dim(i).has_dim_value() && B.dim(i).has_dim_value()) {
-        if (A.dim(i).dim_value() != B.dim(i).dim_value()) {
-        fail_check("The compared dim values are different.(",
-          A.dim(i).dim_value(), ") vs (", B.dim(i).dim_value(), ").");
-        return false;
-        }
-      } else if (A.dim(i).has_dim_param() && B.dim(i).has_dim_param()) {
-        if (checkSameParam && A.dim(i).dim_param() != B.dim(i).dim_param()) {
-          fail_check("The compared dim parameters are different.(",
-            A.dim(i).dim_param(), ") vs (", B.dim(i).dim_param(), ").");
-          return false;
-        }
-        else {
-          continue;
-        }
-      } else {
-        fail_check("Cannot compare a dim parameter to a dim value.");
-        return false;
-      }
-    }
-    return true;
+inline bool CompareShape(
+    const TensorShapeProto& inferredShape,
+    const TensorShapeProto& expectedShape,
+    bool checkSameParam = false) {
+  EXPECT_TRUE(inferredShape.dim_size() == expectedShape.dim_size(), "Dim size for inferred and expected shape is different.");
+
+  for (int i = 0; i < inferredShape.dim_size(); i++) {
+    EXPECT_TRUE(
+        (inferredShape.dim(i).has_dim_value() && expectedShape.dim(i).has_dim_value()) ||
+            (inferredShape.dim(i).has_dim_param() && expectedShape.dim(i).has_dim_param()),
+        "Inferred and expected dim values are different.");
```

Comment:
Should be `EXPECT_TRUE(
        (inferredShape.dim(i).has_dim_value() && expectedShape.dim(i).has_dim_value()) ||
            (inferredShape.dim(i).has_dim_param() && expectedShape.dim(i).has_dim_param())) <<
        "Inferred and expected dim values are different.";` ?

Reply:
yes updated